src/FreeScribe.client/UI/SettingsWindow.py

```diff
@@ -502,3 +502,28 @@ def _create_settings_and_aiscribe_if_not_exist(self):
             print("AIScribe2 file not found. Creating default AIScribe2 file.")
             with open(get_resource_path('aiscribe2.txt'), 'w') as f:
                 f.write(self.AISCRIBE2)
+
+    def get_available_architectures(self):
+        """
+        Returns a list of available architectures for the user to choose from.
+
+        Based on the install state files in _internal folder
+
+        Files must be named CPU_INSTALL or NVIDIA_INSTALL
+
+        Returns:
+            list: A list of available architectures for the user to choose from.
+        """
+
+        # Check the internal folder for state file
+        # Either CPU_INSTALL or NVIDIA_INSTALL file
+        # Can add more in the future like METAL_INSTALL
+
+        if os.path.isfile(get_file_path("state_files", "CPU_INSTALL.txt")):
+            return ["CPU"]
+        elif os.path.isfile(get_file_path("state_files", "NVIDIA_INSTALL.txt")):
```

Comment:
Basically impossible for this to happen but (I'm sure it would happen somehow still haha) I guess ill throw in a case where both exist!!

Added here: 8e131232f63abdfec84fc3a42f2d9704c701629e